src/main/java/edu/ucar/unidata/rosetta/service/EmbeddedDerbyDbInitManager.java

```diff
@@ -0,0 +1,218 @@
+package edu.ucar.unidata.rosetta.service;
+
+import java.io.File;
+
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import java.util.Properties;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+
+import org.springframework.dao.NonTransientDataAccessResourceException;
+
+/**
+ * An implementation of the DbInitManager that creates
+ * the default embedded derby database for the rosetta application.
+ *
+ * @author oxelson@ucar.edu
+ */
+public class EmbeddedDerbyDbInitManager implements DbInitManager {
+
+    private static final Logger logger = Logger.getLogger(EmbeddedDerbyDbInitManager.class);
+
+
+    /**
+     * Creates the embedded derby database for the rosetta.
+     *
+     * @param props RosettaProperties used to create the database.
+     * @throws NonTransientDataAccessResourceException  If unable to create or access the database.
+     * @throws SQLException If an SQL exceptions occurs during database creation.
+     */
+    @Override
+    public void createDatabase(Properties props) throws NonTransientDataAccessResourceException, SQLException {
+
+        // Get relevant properties.
+        String rosettaHome = props.getProperty("rosetta.home");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "rosetta.home" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)